sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/InstrumentSelector.java

```diff
@@ -30,45 +35,30 @@ public static Builder newBuilder() {
    * be used.
    */
   @Nullable
-  public abstract InstrumentType instrumentType();
-
-  /**
-   * Returns which instrument names should be selected. This is a regex. If null, then this
-   * specifier will not be used.
-   */
-  @Nullable
-  public abstract String instrumentNameRegex();
+  public abstract InstrumentType getInstrumentType();
 
   /**
-   * Returns the {@link Pattern} generated by the provided {@link #instrumentNameRegex()}, or null
-   * if none was specified.
+   * Returns the {@link Pattern} generated by the provided {@code regex} in the {@link Builder}, or
+   * null if none was specified.
    */
   @Nullable
-  @Memoized
-  public Pattern instrumentNamePattern() {
-    return instrumentNameRegex() == null ? null : Pattern.compile(instrumentNameRegex());
-  }
-
-  /** Returns whether the InstrumentType been specified. */
-  public boolean hasInstrumentType() {
```

Comment:
I think it reads a little better to have this helper methods, rather than requiring callers to look for nulls.

Reply:
Helper public methods means more to maintain and take care of, there is only one usage of this and I think it is worth the downside of reading to not have to maintain ABI compatibility.